waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -69,8 +84,57 @@ analyzeWaspProject waspDir options = do
   where
     fileNotFoundMessage = "Couldn't find the *.wasp file in the " ++ toFilePath waspDir ++ " directory"
 
-analyzeWaspFile :: Psl.Schema.Schema -> Path' Abs File' -> IO (Either [CompileError] [AS.Decl])
-analyzeWaspFile prismaSchemaAst waspFilePath = do
+analyzeWaspFile :: Psl.Schema.Schema -> WaspFile -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspFile prismaSchemaAst = \case
+  WaspLangFile waspFilePath -> analyzeWaspLangFile prismaSchemaAst waspFilePath
+  WaspTsFile waspFilePath -> analyzeWaspTsFile prismaSchemaAst waspFilePath
+
+analyzeWaspTsFile :: Psl.Schema.Schema -> Path' Abs File' -> IO (Either [CompileError] [AS.Decl])
+analyzeWaspTsFile _prismaSchemaAst waspFilePath = do
+  let workingDir = SP.parent waspFilePath
+  chan <- newChan
+  (_, tscExitCode) <-
+    concurrently
+      (readJobMessagesAndPrintThemPrefixed chan)
+      (runNodeCommandAsJob workingDir "npx" ["tsc", "-p", "tsconfig.node.json"] J.Wasp chan)
+
+  case tscExitCode of
+    ExitFailure _status -> return $ Left ["Error while running TypeScript compiler on the *.wasp.mts file."]
+    ExitSuccess -> do
+      otherChan <- newChan
+      (_, runExitCode) <-
+        concurrently
+          (handleRunJsMessages otherChan)
+          ( runNodeCommandAsJob
+              workingDir
+              "node"
+              [ SP.fromAbsDir workingDir ++ "node_modules/wasp-ts-sdk/dist/run.js",
+                SP.fromAbsDir workingDir ++ ".wasp/config/main.wasp.mjs"
+              ]
+              J.Wasp
+              otherChan
+          )
+      case runExitCode of
+        ExitFailure _status -> return $ Left ["Error while running the compiled *.wasp.mts file."]
+        ExitSuccess -> return $ Right []
+  where
+    handleRunJsMessages :: Chan J.JobMessage -> IO ()
+    handleRunJsMessages = runPrefixedWriter . processMessages
+    processMessages chan = do
+      jobMsg <- liftIO $ readChan chan
+      case J._data jobMsg of
+        J.JobOutput payload J.Stdout -> do
+          -- let payload:: String = read $ T.unpack text
+          liftIO $ putStrLn "Ovo smo dobili na stdout"
+          -- parse payload as json
+          let json = Aeson.toJSON payload
```

Comment:
Aha this is still WIP! Sorry I thought this PR is ready for review since it wasn't draft PR.

Reply:
Aw man, I made a git error.
The `Analyze.hs` file was never supposed to be here, must have committed it by accident.

Sorry for wasting your time, my bad!

What worries me is that you got through the entire review without noticing that the code is too horrible for something I'd send to code review.

Commenting with a straight face at all the atrocities and only realizing something was off when you encountered a string in Croatian.

Is that what you think of me, Martin? 

Anyway, I'll remove this file from the PR. Please take a look at the other files. :)

Reply:
Hah, I believed in you!

All right, I will take another look. Maybe of some the change syou did in this file though, and that I commented on, are still present in the version of the code you are woroking on? In that case do take a look at my comments on Analyze.hs and see if you can apply them.

Reply:
Or maybe this is just your sneaky way of giving up on a bad PR :D?